/* Copyright 2020 Istio Authors. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

// DO NOT MODIFY.
// This is a copy of envoy-wasm proto, used for Wasm build only.

syntax = "proto3";

package envoy.source.extensions.common.wasm;

enum WasmType {
  Bytes = 0;
  String = 1;
  FlatBuffers = 2;
  Protobuf = 3;
};

enum LifeSpan {
  FilterChain = 0;
  DownstreamRequest = 1;
  DownstreamConnection = 2;
};

message DeclarePropertyArguments {
  string name = 1;
  bool readonly = 2;
  WasmType type = 3;
  bytes schema = 4;
  LifeSpan span = 5;
};
